### ISSUE TITLE
Potential bug - Usage of from_logits=True on the model that has softmax activated output

### ISSUE THREAD

**Issue by ksachdeva** [mmalekzadeh/dana#1]
Hi,

First of all, thank you for an excellent paper and also for developing a very good corresponding code base.

I tried to run your notebook in the google colab and have observed few issues

```
def Ordonez2016DeepOriginal(inp_shape, out_shape):   
    nb_filters = 64 
    drp_out_dns = .5 
    nb_dense = 128 
    
    inp = Input(inp_shape)

    x = Conv2D(nb_filters, kernel_size = (5,1),
              strides=(1,1), padding='valid', activation='relu')(inp)    
    x = Conv2D(nb_filters, kernel_size = (5,1),
              strides=(1,1), padding='valid', activation='relu')(x)
    x = Conv2D(nb_filters, kernel_size = (5,1), 
              strides=(1,1), padding='valid', activation='relu')(x)
    x = Conv2D(nb_filters, kernel_size = (5,1), 
              strides=(1,1), padding='valid', activation='relu')(x)    
    x = Reshape((x.shape[1],x.shape[2]*x.shape[3]))(x)
    act = LSTM(nb_dense, return_sequences=True, activation='tanh', name="lstm_1")(x)        
    act = Dropout(drp_out_dns, name= "dot_1")(act)
    act = LSTM(nb_dense, activation='tanh', name="lstm_2")(act)        
    act = Dropout(drp_out_dns, name= "dot_2")(act)
    out_act = Dense(out_shape, activation='softmax',  name="act_smx")(act)
    
    model = keras.models.Model(inputs=inp, outputs=out_act)
    return model

def standard_training(model, X_train, Y_train, X_val, Y_val, data_class_weights,
                      batch_size=128, num_epochs=128, save_dir=None):
    loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
```

Since your model's last layer i.e. `out_act` is using softmax, I think you should use 
```
loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=False)
```

As such now Tensorflow 2.6 is generating a warning to arrest this kind of situation.

I changed it and the testing accuracy on the standard model is 0.9213 and training loss is zero.... but of course, there is always an element of randomness here.

This issue of `from_logits=True` is also present when you use the DANA model.

However, I am not able to run the DANA model because of another problem. I am creating a separate issue for that.

Regards & thanks
Kapil

**Comment by mmalekzadeh** [comment]
Many thanks @ksachdeva 

Yes. You are right We should use `from_logits=False`. I have updated the notebook and now this is resolved.
Unfortunately, TF 2.3.0 did not generate such a warning. 

The result after this change (in terms of the test accuracy) is improved a bit for  both models. 
Using `from_logits=True` was like applying softmax twice. 
But, as this was the same for both models, the relative results did not changed. 

Cheers,
Mohammad

**Comment by ksachdeva** [comment]
Thanks, @mmalekzadeh for the quick response.
